tests/test_helpers.py

```diff
@@ -150,6 +150,24 @@ def test_deserialize_valid_6(self) -> None:
             datetime(year=2001, month=10, day=26, hour=21, minute=32, second=52, microsecond=12679, tzinfo=None)
         )
 
+    def test_deserialize_valid_7(self) -> None:
```

Comment:
you tests showcase only the 6 digit microseconds. according to your description, this case is already working.

please add regression test for the feature you tryed to enable

Reply:
Hi, I appreciate you taking a look at this so quickly!

The existing tests in the library already cover less than 6 digits and my added tests cover exactly 6 digits and more than 6 digits, what extra tests would you like me to add?

Reply:
Oh, i did not recognize the case of your tests.
could you add comments to them, describing what they aim for?

Reply:
I have added some docstrings, let me know if you want me to be more specific in my descriptions.

Reply:
Hi, I have improved the added tests so they now validate that the regex has either correctly padded, not touched, or truncated the decimals in the seconds field. This allows for the tests to fail on newer Python versions if there is a regression (since Python 3.11's datetime works with any number of decimals so if the padding/truncation broke they would previously only fail on 3.10 and earlier).

Reply:
thank you so much